### PR TITLE
*: remove `portgenerator` to cleanup codes

### PR DESCRIPTION
**Pull request by bb7133**
### What problem does this PR solve?

Issue Number: Not needed, since this is not related to 'core codes' of TiDB.

Problem Summary:

### What changed and how does it work?

The package `portgenerator` was introduced in https://github.com/pingcap/tidb/pull/18951. Allocating the available ports rather than using 'fixed' ports is a good idea. However, I don't think it's a good idea to write some Go code, build it, and call it in the shell script.

Instead, I introduced `find_multiple_available_ports()` in `run-tests.sh`, which achieved basically the same goal.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No need to test
  > - [x] I checked and no code files have been changed.
  > 

Side effects

- NA

Documentation

- NA

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: tests/integrationtest/run-tests.sh [hunk 1]
```diff
@@ -58,19 +58,54 @@ function help_message()
...
-    echo "building portgenerator binary: $portgenerator"
-    rm -rf $portgenerator
-    GO111MODULE=on go build -o $portgenerator github.com/pingcap/tidb/cmd/portgenerator
+# Function to find an available port starting from a given port
+function find_available_port() {
+    local start_port=$1
+    local port=$start_port
+
+    while :; do
+        if command -v ss &> /dev/null; then
```
Comment (Defined2014) [inline tests/integrationtest/run-tests.sh:70]: Does our docker image has `ss` command?

Comment (Defined2014) [reply tests/integrationtest/run-tests.sh:70]: Maybe we could read `/proc/net/tcp` file to get it.

Comment (bb7133) [reply tests/integrationtest/run-tests.sh:70]: I'll use `lsof` instead.

Comment (Defined2014) [reply tests/integrationtest/run-tests.sh:70]: Not sure if our docker has it

File: tests/integrationtest/run-tests.sh [hunk 2]
```diff
@@ -58,19 +58,42 @@ function help_message()
...
+    local ports=()
+
+    while [ ${#ports[@]} -lt $count ]; do
+        local available_port=$(find_available_port $start_port)
+        if [ $? -eq 0 ]; then
+            ports+=($available_port)
+            ((start_port = available_port + 1))
+        else
+            echo "Error: Could not find an available port." >&2
+            return 1
```
Comment (Defined2014) [inline tests/integrationtest/run-tests.sh:91]: should be `exit 1`

Comment (bb7133) [reply tests/integrationtest/run-tests.sh]: Addressed, PTAL

File: tests/integrationtest/run-tests.sh [hunk 3]
```diff
@@ -58,19 +58,42 @@ function help_message()
...
+function find_available_port() {
+    local port=$1
+
+    while :; do
+        # Use lsof to check if the port is in use
+        if ! lsof -i :"$port" &> /dev/null; then
+            echo $port
+            return 0
+        fi
+        ((port++))
```
[the remark this answers is not in the document]
Comment (bb7133) [reply tests/integrationtest/run-tests.sh:77]: Addressed, PTAL

**Review by Defined2014 — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**